projects/control-service/projects/pipelines_control_service/src/integration-test/java/com/vmware/taurus/datajobs/it/common/BaseDataJobDeploymentIT.java

```diff
@@ -38,4 +49,35 @@ public TaskExecutor taskExecutor() {
          return new SyncTaskExecutor();
       }
    }
+
+   protected MvcResult executeDataJob(String jobName, String teamName, String username, String opId) {
+      // Execute data job
+      if (opId == null) {
+         opId = jobName + UUID.randomUUID().toString().toLowerCase();
+      }
+
+      DataJobExecutionRequest dataJobExecutionRequest = new DataJobExecutionRequest()
+            .startedBy(username);
+
+      String triggerDataJobExecutionUrl = String.format(
+            "/data-jobs/for-team/%s/jobs/%s/deployments/%s/executions",
+            teamName,
+            jobName,
+            "release");
+
+      // Wait for the job execution to complete, polling every 15 seconds
+      // See: https://github.com/awaitility/awaitility/wiki/Usage
+      String finalOpId = opId;
+      return await()
+            .atMost(5, TimeUnit.MINUTES)
+            .with()
+            .pollInterval(15, TimeUnit.SECONDS)
+            .until(() -> mockMvc.perform(post(triggerDataJobExecutionUrl)
```

Comment:
Shouldn't you execute the job first, then wait for the completion?

Reply:
Maybe `synchronization` is not the right term. The data job execution may not have been completed by the previous test and the API may return 409 instead of 202. This method waits for data job execution completion and then starts a new one.